src/parser/jobs/sch/modules/Interrupts.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import {Table, Button} from 'semantic-ui-react'
+import {Trans} from '@lingui/react'
+import {t} from '@lingui/macro'
+
+import Module from 'parser/core/Module'
+import {getDataBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import {ActionLink} from 'components/ui/DbLink'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+const TIMELINE_UPPER_MOD = 30000
+
+const INTERRUPT_SEVERITY = {
+	2: SEVERITY.MEDIUM,
+	5: SEVERITY.MAJOR,
+}
+
+export default class Interrupts extends Module {
+	static handle = 'interrupts'
+	static title = t('sch.interrupts.title')`Interrupted Casts`
+
+	static dependencies = [
+		'timeline',
+		'gcd',
+		'suggestions',
+	]
+
+	_currentCast = null
+	_droppedCasts = []
+	_missedTimeMS = 0
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('begincast', {by: 'player'}, this._onBeginCast)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onBeginCast(event) {
+		// if they started casting something, then cancel and start something else, that's definitely an interrupted cast
+		if (this._currentCast) {
+			this._pushDropCasts(event)
+		}
+		this._currentCast = event
+	}
+
+	_onCast(event) {
+		const guid = event.ability.guid
+		// if the thing they started casting doesn't match up with what they cast, that's an interrupted cast
+		if (this._currentCast && guid !== this._currentCast.ability.guid) {
+			this._pushDropCasts(event)
+		}
+		this._currentCast = null
+	}
+
+	_pushDropCasts(currentEvent) {
+		this._missedTimeMS += Math.min(this.gcd.getEstimate(), currentEvent.timestamp - this._currentCast.timestamp)
+		this._droppedCasts.push({
+			action: getDataBy(ACTIONS, 'id', this._currentCast.ability.guid),
+			timestamp: this._currentCast.timestamp,
+		})
+	}
+
+	_onComplete() {
+		this.suggestions.add(new TieredSuggestion({
+			icon: ACTIONS.INTERJECT.icon,
+			tiers: INTERRUPT_SEVERITY,
+			value: this._droppedCasts.length,
+			content: <Trans id="sch.interrupts.suggestion.content">If you can, try to preposition yourself so you don't have to move during mechanics as much as possible. Utilizing slidecasting will lower the need to use <ActionLink {...ACTIONS.SCH_RUIN_II}/> to instantly relocate or interrupt your current Broil III cast</Trans>,
```

Comment:
Yeah, I feel this & the `why` could do with a bit of work, but that can be done at a later stage.

---

nit; avoid adding spaces around substitutions in jsx, looks a bit jank.